packages/engine/src/scene/components/AudioAnalysisComponent.ts

```diff
@@ -0,0 +1,138 @@
+/*
+CPAL-1.0 License
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/EtherealEngine/etherealengine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Ethereal Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import {
+  defineComponent,
+  getComponent,
+  setComponent,
+  useComponent,
+  useEntityContext,
+  useOptionalComponent
+} from '@etherealengine/ecs'
+import { TransformComponent } from '@etherealengine/spatial'
+import { NameComponent } from '@etherealengine/spatial/src/common/NameComponent'
+import { VisibleComponent } from '@etherealengine/spatial/src/renderer/components/VisibleComponent'
+import { useEffect } from 'react'
+import { Color, Mesh, MeshStandardMaterial, SphereGeometry } from 'three'
+import { PositionalAudioComponent } from '../../audio/components/PositionalAudioComponent'
+import { AudioNodeGroups, MediaComponent, MediaElementComponent } from './MediaComponent'
+
+export const AudioAnalysisComponent = defineComponent({
+  name: 'EE_audioAnalyzer',
+  jsonID: 'audio-analyzer',
+
+  onInit: (entity) => {
+    return {
+      src: '',
+      dataArray: null as Uint8Array | null,
+      analyser: null as AnalyserNode | null,
+      bassEnabled: true as boolean,
+      midEnabled: true as boolean,
+      trebleEnabled: true as boolean,
+      bassMultiplier: 1 as number,
+      midMultiplier: 1 as number,
+      trebleMultiplier: 1 as number
+    }
+  },
+
+  onSet: (entity, component, json) => {
+    if (!json) return
+    if (json.src && component.src.value !== json.src) {
+      component.src.set(json.src)
+    }
+    if (json.dataArray && component.dataArray.value !== json.dataArray) {
+      component.dataArray.set(json.dataArray)
+    }
+    if (json.bassEnabled) {
```

Comment:
we need to assert `typeof json.property === number` or use `matches.number.test(json.property` otherwise falsy values will not be allowed to be set